### PR TITLE
Add Sum to TStatistic

### PR DESCRIPTION
**Pull request by vepadulano**
The sum of all the values passed to fill the `TStatistic` object has been added as a class data member. Meanwhile, `Fill`,`Print`, and `Merge` functions have been changed to accomodate the new statistic. The sum is initialized to zero when calling the constructor of TStatistic. Docs and test have been updated
accordingly.

### CODE REVIEW

File: math/mathcore/src/TStatistic.cxx [hunk 1]
```diff
@@ -56,80 +60,90 @@ TStatistic::~TStatistic()
...
+   if (fW != 0) {  // From the second time
       Double_t rr = ( tW * val - fM);
       fM2 += w * rr * rr / (tW * fW);
    }
+   // Update sum of weights in the TStatistic object
    fW = tW;
+   // Update sum of weights squared
    fW2 += w*w;
+   // Update sum of values
+   fSum += val;
```
Comment (dpiparo) [inline math/mathcore/src/TStatistic.cxx]: do you really want to ignore the weight?

Comment (vepadulano) [reply math/mathcore/src/TStatistic.cxx]: There is already the weighted sum in the class data members, it's just not printed. Maybe I could refactor the PR in order to print the weighted sum when the user also includes weights as arguments to the constructor, or just the simple sum if there are only the values

File: math/mathcore/inc/TStatistic.h [hunk 2]
```diff
@@ -43,10 +43,13 @@ class TStatistic : public TObject {
    Double_t    fM2;      ///< Second order momentum
    Double_t    fMin;     ///< Minimum value in the Tstatistic object
    Double_t    fMax;     ///< Maximum value in the TStatistic object
+   Double_t    fSum;     ///< Sum of the values in the TStatistic object
 
 public:
 
-   TStatistic(const char *name = "") : fName(name), fN(0), fW(0.), fW2(0.), fM(0.), fM2(0.), fMin(TMath::Limits<Double_t>::Max()), fMax(TMath::Limits<Double_t>::Min()) { }
+   TStatistic(const char *name = "")
+      : fName(name), fN(0), fW(0.), fW2(0.), fM(0.), fM2(0.), fSum(0.)
```
Comment (Axel-Naumann) [inline math/mathcore/inc/TStatistic.h]: Can we (in a subsequent PR) not make those in-class initializers?
```
Double_t    fSum{0.};     ///< Sum of the values in the TStatistic object
```
etc. With that we'd be left with `TStatistic() = default` and `TStatistic(const char *name): TStatistic() {...}`.

**Comment by bluehood** [comment]
@dpiparo @vepadulano what should we do with this one?

**Comment by vepadulano** [comment]
The changes to the class still hold valid, I was held back by a strange behaviour in `clang-format`. Will resume work asap, thanks

**Comment by vepadulano** [comment]
@phsft-bot build

**Comment by vepadulano** [comment]
Stale PR, closing for now. Will return on it if needed

**Pull request closed without merging**